[jniaorx] ip

jniaorx opened the pull request:
# IntroBlaise
> "Soon is not as good as now." - Seth Godin ([source](https://dansilvestre.com/productivity-quotes/))

IntroBlaise frees your mind of having to remember things you need to do.
It's,

+ text-based
* easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/jniaorx/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

jniaorx committed:
Renamed Duke and updated logo.

jniaorx committed:
Created a new class Greetings.

jniaorx committed:
Added hi and bye messages.

jniaorx committed:
Added echoed and exit commands.

jniaorx committed:
Modified oop.

jniaorx committed:
Added indentation.

jniaorx committed:
Customised greeting text by adding user's name.

jniaorx committed:
Added tasklist functionality.

jniaorx committed:
Added some comments.

jniaorx committed:
Added the ability to mark or unmark tasks.

jniaorx committed:
Added some comments.

jniaorx committed:
Added Deadline class for tasks with deadline.

jniaorx committed:
Added ToDo class for tasks without deadline.

jniaorx committed:
Added Event class for tasks that starts and ends at a specific date/time.

jniaorx committed:
Fixed mark and unmark bugs.

jniaorx committed:
Modified runtest.bat to do for automating the testing of text UIs.

jniaorx committed:
Added content to input.txt.

jniaorx committed:
Added content to EXPECTED.TXT.

jniaorx committed:
Modified TaskManageer class.

jniaorx committed:
Modified TaskManager class for better oop.

jniaorx committed:
Modified Commands class for better oop.

jniaorx committed:
Modified IntroBlaise class for better oop.

jniaorx committed:
Added a space in the string.

jniaorx committed:
Created a AlreadyUndoneException exception class.

jniaorx committed:
Created a EmptyDescriptionException exception class.

jniaorx committed:
Created an IntroBlaiseException exception class.

jniaorx committed:
Created InvalidDeadlineFormatException exception class.

jniaorx committed:
Created an InvalidEventFromFormatException exception class.

jniaorx committed:
Created an InvalidEventToFormatException exception class.

jniaorx committed:
Created InvalidInputException exception class.

jniaorx committed:
Handled possible exceptions.

jniaorx committed:
Implemented delete task function.

jniaorx committed:
Added exceptions to deleteTask method.

jniaorx committed:
Added EmptyTaskListException exception class.

jniaorx committed:
Modified greetings.

jniaorx committed:
Modified printed strings.

jniaorx committed:
Modified exception messages.

jniaorx committed:
Modified comments in Commands class.

jniaorx committed:
Modified comments in AlreadyUndoneException class.

jniaorx committed:
Modified comments in Deadline class.

jniaorx committed:
Modified comments in Echo class.

jniaorx committed:
Modified comments in EmptyDescriptionException class.

jniaorx committed:
Modified comments in EmptyTaskListException class.

jniaorx committed:
Modified comments in Event class.

jniaorx committed:
Modified comments in Greetings class.

jniaorx committed:
Modified comments in IntroBlaise class.

jniaorx committed:
Modified comments in IntroBlaiseException class.

jniaorx committed:
Modified comments in InvalidDeadlineFormatException class.

jniaorx committed:
Modified comments in InvalidEventFromFormatException class.

jniaorx committed:
Modified comments in InvalidEventToFormatException class.

jniaorx committed:
Modified comments in InvalidInputException class.

jniaorx committed:
Modified comments in Task class.

jniaorx committed:
Modified comments in TaskManager class.

jniaorx committed:
Modified comments in ToDo class.

jniaorx committed:
Corrected a typo in ToDo class.

jniaorx committed:
Corrected typos.

jniaorx committed:
Edited comment.

jniaorx committed:
Edited exception string.

jniaorx committed:
Create SaveData class to save tasks in hard disk.

jniaorx committed:
Implement saving tasks in hard disk automatically whenever the task list changes

jniaorx committed:
Modified TaskManager.java to save tasks automatically

jniaorx committed:
Change method saveTasks to private

jniaorx committed:
Implement loading data when chatbot starts

jniaorx committed:
Implement Level 7

jniaorx committed:
Merge barnch-level-7 into master

jniaorx committed:
Add deadline parsing function in Deadline class

jniaorx committed:
Add event duration parsing function in Event class

jniaorx committed:
Handle error where there are no tasks in hard disk when chatbot starts

jniaorx committed:
Add deadline parsing function in Event class

jniaorx committed:
Change date format

jniaorx committed:
Add command to print deadlines/events occuring on a specific date

jniaorx committed:
Merge branch-level-8 into master

jniaorx committed:
Add Ui class

jniaorx committed:
Rename SaveData class to Storage class

jniaorx committed:
Move StringToTask method to TaskManager class

jniaorx committed:
Rename Commands class to Parser class

jniaorx committed:
Rename TaskManager class to TaskList class

jniaorx committed:
Merge branch-A-MoreOOP into master

jniaorx committed:
Organize classes into Packages

jniaorx committed:
merge branch-A-Packages into master

jniaorx committed:
Delete .idea folder

jniaorx committed:
Edit mainClass.set() in build.gradle file

jniaorx committed:
Create ToDoTest class

jniaorx committed:
Create testAddTaskAndRetrieve method to test addTask method

jniaorx committed:
Create TaskTest to test methods in Task class

jniaorx committed:
Merge branch-A-Junit into master

jniaorx committed:
Add JavaDOc comments to Storage.java

jniaorx committed:
Add JavaDoc comments to Deadline.java

jniaorx committed:
Add JavaDoc comments to Event.java

jniaorx committed:
Modified JavaDoc comments on Task.java

jniaorx committed:
Add JavaDoc comments to TaskList.java

jniaorx committed:
Modify JavaDoc comments on ToDo.java

jniaorx committed:
Modify JavaDoc comments on Echo.java

jniaorx committed:
Modify JavaDoc comments on Greetings.java

jniaorx committed:
Modify JavaDoc comments on IntroBlaise.java

jniaorx committed:
Modify JavaDoc comments on Parser.java

jniaorx committed:
Modify JavaDoc comments on Ui.java

jniaorx committed:
Change packages names to lower case

jniaorx committed:
Change test method name in TaskListTest.java

jniaorx committed:
Change test method name in TaskTest.java

jniaorx committed:
Reduce line lengths

jniaorx committed:
List imported classes explicitly

jniaorx committed:
Add find command function

jniaorx committed:
Merge branch-A-JavaDoc into master

jniaorx committed:
Resolve conflicts

jniaorx committed:
Resolve conflicts

limshaoqi commented on src/main/java/introblaise/storage/Storage.java:
```diff
@@ -0,0 +1,113 @@
+ * write tasks to a text file.
+ */
+public class Storage {
+    // Path for the directory where tasks will be stored
+    private static final String DIRECTORY_PATH = "data";
+
+    // Path for the file containing tasks
+    private static final String FILE_PATH = "data/introBlaise.txt";
```
Well done on using constants for the directory and the file paths. This improves maintainability. 
lesgooo skrttt

limshaoqi commented on src/main/java/introblaise/ui/IntroBlaise.java:
```diff
@@ -0,0 +1,38 @@
+/**
+ * The entry point for the IntroBlaise bot.
+ */
+public class IntroBlaise {
```
class name follows PascalCase convention, which is correct. Keep up the good work G!

limshaoqi commented on src/main/java/introblaise/storage/Storage.java:
```diff
@@ -0,0 +1,113 @@
+     *
+     * @return A list of task descriptions read from the file.
+     */
+    public List<String> loadTasks() {
```
loadTasks is clear and descriptive. However, it might be more helpful to rename to 'loadTasksFromFile' to make it even more descriptive. :)

jniaorx committed:
Catch invalid date format error for deadline task

jniaorx committed:
Modify event toString() date format

jniaorx committed:
Catch invalid date format error for event task

jniaorx committed:
Modify EmptyDescriptionException class

moingshoing commented on src/main/java/introblaise/task/TaskList.java:
```diff
@@ -0,0 +1,235 @@
+     * Saves the tasks list to storage.
+     * Converts task objects to strings before saving them.
+     */
+    public void saveTasks() {
```
I like how you name your method which clearly tells me that you're saving the entire list instead of one task

moingshoing commented on src/main/java/introblaise/task/TaskList.java:
```diff
@@ -0,0 +1,235 @@
+        }
+    }
+
+    public List<Task> findTasksByKeyword(String keyword) {
```
Good naming convention that explains the method clearly

moingshoing commented on src/main/java/introblaise/task/Event.java:
```diff
@@ -0,0 +1,84 @@
+ */
+public class Event extends Task {
+    /** The start date/time of the event represented as a string. */
+    protected String from;
+
+    /** The end date/time of the event represented as a string. */
+    protected String to;
```
Good use of "from" and "to" to distinguish between the two Strings

jniaorx committed:
Add main and launcher class for creation of GUI

jniaorx committed:
Add dialog boxes for bot's response

jniaorx committed:
Flip the dialog box such that the bot's output is on the left edge

jniaorx committed:
Refractor IntroBlaise to use FXML

jniaorx committed:
Change printTaskList method to return String

jniaorx committed:
Change markTaskAsDone method to return String

jniaorx committed:
Change addDeadlineTask method to return String

jniaorx committed:
Change deleteTask method to return String

jniaorx committed:
Change getTasksOnDate method to return String

jniaorx committed:
Change handleFindCommand method to return String

jniaorx committed:
Add newline to some of the responses

jniaorx committed:
Add welcome message to the GUI

jniaorx committed:
Add bye command to say bye to user

jniaorx committed:
Add javadoc header to showWelcomeMessage method

jniaorx committed:
Merge branch-level-10 into master

jniaorx committed:
Correct a typo

jniaorx committed:
Add Checkstyle plugin

jniaorx committed:
Edit coding styles based on Checkstyle

jniaorx committed:
Merge branch-A-CheckStyle into master

jniaorx committed:
Modify bye command to close the app

jniaorx committed:
Rename loadTasks method to loadTasksFromFile to make it even more descriptive

jniaorx committed:
Add tasktype package and refractor some of the pacakages

jniaorx committed:
Assertions are currently not used in key functions, making it hard to
verify assumptions.

This needs to change because without assertions, incorrect assumptions
may go unnoticed, leading to unexpected behaviour.

Add assertions to critical functions to enforce expected conditions and
document assumptions.

This approach is chosen because assertions help in validating
constraints during developing and testing, without affecting performance
in production if assertions are disabled.

jniaorx committed:
The expression if (eventTask.getParsedFrom() == null ||
eventTask.getParsedTo() == null) is slightly long and complicated.

Splitting them into a few steps helps to improve readability.

This change improves the code quality and follows the principle of
writing clear and maintanable code, ensuring logical conditions are easy
to follow.

jniaorx committed:
Merge pull request #1 from jniaorx/branch-A-Assertions

Add assertions to the code.

jniaorx committed:
Merge branch 'master' into branch-A-CodeQuality

jniaorx committed:
Merge pull request #2 from jniaorx/branch-A-CodeQuality

Improve the code quality

jniaorx committed:
Add a tagging extension

jniaorx committed:
Merge branch-BCD-Extension into master

jniaorx committed:
Change bot profile picture

jniaorx committed:
Refractor code to use switch-case instead of if-else

jniaorx committed:
Refractor CommandParser class

jniaorx committed:
Add try-catch in some methods

jniaorx committed:
Improve the UI of the chatbot

jniaorx committed:
Merge branch-A-BetterGui into master

jniaorx committed:
Add app title

jniaorx committed:
Fix StorageTaskParser bug

jniaorx committed:
Improve UI for welcomeLabel

jniaorx committed:
Reformat code for better coding quality

jniaorx committed:
Add user guide for the bot

jniaorx committed:
Merge branch-A-UserGuide into master

jniaorx committed:
Handle more exceptions

jniaorx committed:
Merge branch-A-MoreErrorHandling into master

jniaorx committed:
Add more JUnit tests

jniaorx committed:
Add more JUnit tests

jniaorx committed:
Update REAME.md

jniaorx committed:
Create a new jar file for release

jniaorx committed:
Merge branch into master

jniaorx committed:
Fix findTasksOn bug

jniaorx committed:
Update Ui

jniaorx committed:
Update README.md

jniaorx committed:
Edit javadoc comments

jniaorx committed:
Fix parser bug

jniaorx committed:
Generate a new jar file
